gallifreyan.js

```diff
@@ -72,15 +75,29 @@ function updateText() {
     generateWords(words);
 }
 
+//disconnect the line from both circles it's connected to, and remove it from the global list
+function deleteLine(line) {
+    for (var c = 0; c < allCircles.length; c++) {

```

Comment:
A line holds a list of the circles it's attached to (`line.points`), so you don't have to iterate all of them.